File: Feel@Home/Feel@Home/src/main/java/club/frickel/feelathome/Effect.java
package club.frickel.feelathome;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.util.Map;

/**
 * User: Peter Vollmer
 * Date: 2/26/13
 * Time: 1:32 PM
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Effect implements Serializable{

    @JsonProperty("Description")
    private String description;
    @JsonProperty("Name")
    private String name;
    @JsonProperty("Config")
    private Map<String,Object> config;

    /**
     *
     */
    public Effect() {

    }

    /**
     * @param name
     * @param description
     * @param config
     */
    public Effect(String name, String description, Map<String,Object> config) {
        this.name = name;
        this.description = description;
        this.config = config;
    }

    /**
     * @return
     */
    @JsonProperty("Config")
    public Map<String,Object> getConfig() {
        return config;
    }

    /**
     * @param config
     */
    @JsonProperty("Config")
    public void setConfig(Map<String,Object> config) {
        this.config = config;
    }

    /**
     * @return
     */
    @JsonProperty("Description")
    public String getDescription() {
        return description;
    }

    /**
     * @param description
     */
    @JsonProperty("Description")
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return
     */
    @JsonProperty("Name")
    public String getName() {
        return name;
    }

    /**
     * @param name
     */
    @JsonProperty("Name")
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return
     */
    @Override
    public String toString() {
        return this.name;

    }
}

File: Feel@Home/Feel@Home/src/main/java/club/frickel/feelathome/Effect.kt
package club.frickel.feelathome

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
class Effect : Serializable {

    /**
     * @return
     */
    /**
     * @param description
     */
    @JsonProperty("Description")
    @get:JsonProperty("Description")
    @set:JsonProperty("Description")
    var description: String = ""
    /**
     * @return
     */
    /**
     * @param name
     */
    @JsonProperty("Name")
    @get:JsonProperty("Name")
    @set:JsonProperty("Name")
    var name: String = ""
    /**
     * @return
     */
    /**
     * @param config
     */
    @JsonProperty("Config")
    @get:JsonProperty("Config")
    @set:JsonProperty("Config")
    var config: Map<String, Any> = emptyMap()

    /**

     */
    constructor() {

    }

    /**
     * @param name
     * *
     * @param description
     * *
     * @param config
     */
    constructor(name: String, description: String, config: Map<String, Any>) {
        this.name = name
        this.description = description
        this.config = config
    }

    /**
     * @return
     */
    override fun toString(): String {
        return this.name
    }
}